src/animation/AnimationMixer.js

```diff
@@ -199,42 +315,139 @@ THREE.AnimationMixer.prototype = {
+		for ( var i = 0, n = actions.length; i !== n; ++ i ) {
 
-			var actionTimeScale = action.getTimeScaleAt( this.time );
-			var actionDeltaTime = mixerDeltaTime * actionTimeScale;
+			var action = actions[ i ];
+			if ( ! action.enabled ) continue;
 
-			var actionResults = action.update( actionDeltaTime );
+			var weight = action.getWeightAt( time );
+			if ( weight <= 0 ) continue;
 
-			if ( action.weight <= 0 || ! action.enabled ) continue;
+			var actionTimeScale = action.getTimeScaleAt( time );
+			var actionTime = action.updateTime( mixerDeltaTime * actionTimeScale );
 
-			for ( var j = 0; j < actionResults.length; j ++ ) {
+			var interpolants = action._interpolants;
+			var propertyMixers = action._propertyBindings;
 
-				var name = action.clip.tracks[j].name;
+			for ( var j = 0, m = interpolants.length; j !== m; ++ j ) {
 
-				action.propertyBindings[ j ].accumulate( actionResults[j], weight );
+				interpolants[ j ].getAt( actionTime );
+				propertyMixers[ j ].accumulate( accuIndex, weight );
 
 			}
 
 		}
 
-		// apply to nodes
-		for ( var propertyBindingKey in this.propertyBindingMap ) {
+		if ( this._bindingsDirty ) {
+
+			this._updateBindings();
+			this._bindingsDirty = false;
+
+		}
+
+		var bindings = this._bindings;
+		for ( var i = 0, n = bindings.length; i !== n; ++ i ) {
 
-			this.propertyBindingMap[ propertyBindingKey ].apply();
+			bindings[ i ].apply( accuIndex );
 
 		}
 
 		return this;
 
-	}
+	},
+
+	// releases cached references to scene graph nodes
+	// pass 'true' for 'unbindOnly' to allow a quick rebind at
+	// the expense of higher cost add / removeAction operations
+	releaseCachedBindings: function( unbindOnly ) {
+
+		var bindingsMaps = this._bindingsMaps;
+
+		for ( var rootUuid in bindingsMaps ) {
+
+			var bindingsMap = bindingsMaps[ rootUuid ];
+
+			var mapChanged = false;
+
+			for ( var trackName in bindingsMap ) {
+
+				var propertyMixer = bindingsMap[ trackName ];
+
+				if ( propertyMixer.referenceCount === 0 ) {
+
+					if ( unbindOnly ) {
+
+						propertyMixer.binding.unbind();
+
+					} else {
+
+						delete bindingsMap[ trackName ];
+						mapChanged = true;
+
+					}
+
+				}
+
+			}
+
+			if ( mapChanged ) {
+
+				remove_empty_map: for (;;) {
+
+					// unless not empty...
+					for ( var k in bindingsMap ) break remove_empty_map;
+
+					delete bindingsMaps[ rootUuid ];
+					break;
+
+				}
+
+			}
+
+		}
+
+	},
+
+	_updateBindings: function() {
+
+		var bindingsMaps = this._bindingsMaps,
+			bindings = this._bindings,
+			writeIndex = 0;
+
+		for ( var rootUuid in bindingsMaps ) {
+
+			var bindingsMap = bindingsMaps[ rootUuid ];
+
+			for ( var trackName in bindingsMap ) {
+
+				var propertyMixer = bindingsMap[ trackName ];
+
+				if ( propertyMixer.referenceCount !== 0 ) {
+
+					bindings[ writeIndex ++ ] = propertyMixer;
+
+				}
+
+			}
+
+		}
+
+		bindings.length = writeIndex;
+
+	},
+
+	_Up: Float64Array.of( 0, 1 ),
+	_Down: Float64Array.of( 1, 0 ),
```

Comment:
I'd maybe add the term "Keys" or "Values" to these so it is a little more obvious what they are.  Also Float32 should be sufficient for everything dealing with animation.